plugin/completion.py

```diff
@@ -130,6 +129,12 @@ def __init__(self, view: sublime.View) -> None:
         self.enabled = False
         self._request_in_flight = False
 
+    def __del__(self) -> None:
```

Comment:
This most likely won't be called on restarting ST but I guess that's acceptable?

Reply:
It does, that's the trick :)

Reply:
It didn't seem like it here.
Trying to write settings from __del__ might be frowned upon as some things are already destroyed.

Reply:
It only removes entries that have a `server` key. If you have multiple entries already you need to remove your `sublime.workspace` cache file if you're using a .sublime-project and otherwise you'll need to remove your global cache file in $DATA/Local.Session.sublime_session next to your license file.

Reply:
Now that I think of it, maybe I was just having wrong expectations. On restarting ST the view is not necessarily destroyed (from user's POV) so maybe it makes sense to preserve that in the session.

What I was testing was:
1) Open JS file with lsp-typescript server enabled
2) Verify that `view.settings().get("auto_complete_triggers")` returns only default triggers
3) Type dot and verify that new entry with `'server': 'lsp-typescript'` was added
4) Restart ST
5) Check `view.settings().get("auto_complete_triggers")`

I'm getting triggers with new item added while I was expecting only default ones.
Closing the view and re-opening shows only default triggers again.